client/src/layout/Header.tsx

```diff
@@ -1,19 +1,55 @@
-import React from "react";
+import React, { useEffect } from "react";
 import { Link, useNavigate } from "react-router-dom";
-import { logout } from "../api/auth";
-import { useMutation } from "@tanstack/react-query";
+// import { logout } from "../api/auth";
+// import { useMutation } from "@tanstack/react-query";
+import { useState } from "react";
+import { API_URL } from "../api/config";
+
 
 type HeaderProps = {
   isLoggedIn: boolean;
 };
 
-export const Header: React.FC<HeaderProps> = ({ isLoggedIn }) => {
-  const logoutMutation = useMutation({ mutationFn: logout });
+export const Header: React.FC<HeaderProps> = ({ }) => {
+  // const logoutMutation = useMutation({ mutationFn: logout });
   const navigate = useNavigate();
+  const [isLoggedIn, setIsLoggedIn] = useState(false);
+
+  useEffect(() => {
+    if (localStorage.getItem("token")) {
+      setIsLoggedIn(true);
+    } else {
+      setIsLoggedIn(false);
+    }
+  }, []);
+
+  useEffect(() => {
+    if (!isLoggedIn) {
+      navigate("/login");
+    }
+  }
+  , [isLoggedIn]);
+
+  const onLogout = async () => {
+    try {
+      const response = await fetch(`${API_URL}/accounts/logout/`, {
```

Comment:
can we use the react query library? you can make these edits in the logout function

Reply:
like axios right? I'll do it asap asw, I was just not sure where my login was broken so was changing to what I've learned. Should've changed it back after checking, my bad
Thanks!